_examples/golang-basics/example.gen.go

```diff
@@ -28,12 +28,50 @@ func WebRPCVersion() string {
 
 // Schema version of your RIDL schema
 func WebRPCSchemaVersion() string {
-	return "v0.0.1"
+	return "golang-basics@v0.0.1"
 }
 
 // Schema hash generated from your RIDL schema
 func WebRPCSchemaHash() string {
-	return "3a3a4087149f1118f0b9494fa254afab46564a1b"
+	return "bd2f9fc6bee7f200ba8b56fe1f287193bc5208de"
+}
+
+type WebRPCGenVersions struct {
+	WebRPCGenVersion string
+	TmplTarget       string
+	TmplVersion      string
+	SchemaVersion    string
+}
+
+// WebRPCHeader
+// TmplVersion might be empty when generated from local template folder
+func WebRPCHeader() string {
+	return "webrpc@v0.19.3;gen-golang@unknown;golang-basics@v0.0.1"
+}
+
+func ParseWebRPCGenVersions(header string) (*WebRPCGenVersions, error) {
+	versions := strings.Split(header, ";")
+
+	if len(versions) != 3 {
+		return nil, fmt.Errorf("expected 3 parts while parsing webrpc header: %s", header)
+	}
+
+	_, webrpcGenVersion, ok := strings.Cut(versions[0], "@")
+	if !ok {
+		return nil, fmt.Errorf("webrpc gen version could not be parsed from: %s", header)
+	}
+
+	tmplTarget, tmplVersion, ok := strings.Cut(versions[1], "@")
+	if !ok {
+		return nil, fmt.Errorf("tmplTarget and tmplVersion could not be parsed from: %s", header)
+	}
+
```

Comment:
We can also parse the name and schema version.
```suggestion
name, schemaVersion, ok := strings.Cut(versions[2], "@")
	if !ok {
		return fmt.Errorf("name and schemaVersion could not be parsed from: %s", header)
	}
```

This would work even with empty name.

What about building the schema in this way, when the RIDL looks like this:
```
name = example          # name of your backend app
version = v0.0.1        # version of your schema
```
`example@v0.0.1`

or for this RIDL without version defined use hash:
```
name = example          # name of your backend app
# version = v0.0.1        # version of your schema
```
`example@3a3a4087149f1118f0b9494fa254afab46564a1b`

Reply:
This makes a lot of sense.. :) Done